packages/main/src/Tree.ts

```diff
@@ -319,6 +347,92 @@ class Tree extends UI5Element {
 		return !!this.header.length;
 	}
 
+	_ondragenter(e: DragEvent) {
+		e.preventDefault();
+	}
+
+	_ondragleave(e: DragEvent) {
+		if (e.relatedTarget instanceof Node && this.shadowRoot!.contains(e.relatedTarget)) {
+			return;
+		}
+
+		this.dropIndicatorDOM!.targetReference = null;
+	}
+
+	_ondragover(e: DragEvent) {
+		const draggedElement = DragRegistry.getDraggedElement();
+		const allItemsTraversed: Array<TreeItemBase> = []; // use the actual items for rearranging
+		const allLiNodesTraversed: Array<HTMLElement> = []; // use the only <li> nodes to determine positioning
+		if (!(e.target instanceof HTMLElement) || !draggedElement) {
+			return;
+		}
+
+		this.walk(item => {
+			allItemsTraversed.push(item);
+			allLiNodesTraversed.push(item.shadowRoot!.querySelector("li")!);
+		});
+
+		const closestPosition = findClosestPosition(
+			allLiNodesTraversed,
+			e.clientY,
+			Orientation.Vertical,
+		);
+
+		if (!closestPosition) {
+			this.dropIndicatorDOM!.targetReference = null;
+			return;
+		}
+
+		let placements = closestPosition.placements;
+
+		closestPosition.element = allItemsTraversed[allLiNodesTraversed.indexOf(closestPosition.element)];
+
+		if (closestPosition.element === draggedElement) {
+			placements = placements.filter(placement => placement !== MovePlacement.On);
+		}
+
+		const placementAccepted = placements.some(placement => {
+			const beforeItemMovePrevented = !this.fireEvent<TreeMoveEventDetail>("move-over", {
+				source: {
+					element: draggedElement,
+				},
+				destination: {
+					element: closestPosition.element,
+					placement,
+				},
+			}, true);
+
+			if (beforeItemMovePrevented) {
+				e.preventDefault();
+				this.dropIndicatorDOM!.targetReference = closestPosition.element;
+				this.dropIndicatorDOM!.placement = placement;
+				return true;
+			}
+
+			return false;
+		});
+
+		if (!placementAccepted) {
+			this.dropIndicatorDOM!.targetReference = null;
+		}
+	}
+
+	_ondrop(e: DragEvent) {
+		e.preventDefault();
+
+		this.fireEvent<TreeMoveEventDetail>("move", {
+			source: {
+				element: DragRegistry.getDraggedElement()!,
+			},
+			destination: {
+				element: this.dropIndicatorDOM!.targetReference!,
+				placement: this.dropIndicatorDOM!.placement,
+			},
+		});
+
```

Comment:
We decided to handle focus automatically, so here call:
`DragRegistry.getDraggedElement()!.focus()`